File: sorting/python/bubble_sort.py
# See documentation.needs.management/index.php/Sorting_Algorithms for
# the algorithm followed.


def bubble_sort(x):
    """
    Bubble sort implemented in python
    """
    x = x[:]
    swaps = True

    while swaps:
        swaps = False
        for i in range(0, len(x)-1):
            if x[i] > x[i+1]:
                # Swap the items at these indexes
                temp = x[i+1]
                x[i+1] = x[i]
                x[i] = temp
                swaps = True

    return x


x = [7, 3, 1, 0, 77, 99, -123, 6543, -0]
y = bubble_sort(x)
print(y)

File: sorting/ruby/bubble_sort.rb
# Ruby implementation of bubble sort.
# See documentation.needs.management/index.php/Sorting_Algorithms for
# the algorithm followed.

def bubble_sort(x)
    x = x[0,x.length()]
    x_range = *(0 ... x.length()-1)
    swapped = true
    while swapped
        swapped = false
        for y in x_range
            if x[y] > x[y+1]
                temp = x[y]
                x[y] = x[y+1]
                x[y+1] = temp
                swapped = true
            end
        end
    end
    return x
end
